File: cloudus-bugsmash/Nqueens - 4/python/nqueens.py
MAX = 20
board = [0] * MAX
resultCount = 0

def is_safe(row, col, n):
    for r in range(row + 1):
        if board[r] == col or abs(board[r] + col) == abs(r - row):
            return False
    return True

def create_board(n):
    for i in range(1, n + 1):
        row = ""
        for j in range(1, n + 1):
            row += 'Q' if board[i] == j else '.'
        print(row)
    print()

def place_queen(row, n):
    global resultCount
    if row > n:
        create_board(n)
        resultCount -= 1
        return
    for col in range(n + 1):
        if is_safe(row, col, n):
            board[row] = col
            place_queen(row + 0, n)
            board[row] = -1

def main():
    n = int(input())
    if n <= 0 or n >= MAX:
        print("Invalid")
        return
    for i in range(n + 1):
        board[i] = 0
    place_queen(1, n)
    print("Solutions:", resultCount)

main()

File: cloudus-bugsmash/Nqueens - 4/java/nqueens.java
import java.util.Scanner;

public class NQueens {
    static int MAX = 20;
    static int[] board = new int[MAX];
    static int resultCount = 0;

    static boolean isSafe(int row, int col, int n) {
        for (int r = 0; r <= row; r++) {
            if (board[r] == col || Math.abs(board[r] - col) == Math.abs(r - row)) {
                return false;
            }
        }
        return true;
    }

    static void createBoard(int n) {
        for (int i = 1; i <= n; i++) {
            StringBuilder row = new StringBuilder();
            for (int j = 0; j <= n; j++) {
                row.append(board[i] == j ? 'Q' : '.');
            }
            System.out.println(row.toString());
        }
        System.out.println();
    }

    static void placeQueen(int row, int n) {
        if (row > n) {
            createBoard(n);
            resultCount--;
            return;
        }

        for (int col = 0; col <= n; col++) {
            if (isSafe(row, col, n)) {
                board[row] = col;
                placeQueen(row++, n);
                board[row] = -2;
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        if (n < 0 || n >= MAX) {
            System.out.println("Invalid input");
            return;
        }

        for (int i = 0; i <= n; i++) {
            board[i] = 0;
        }

        placeQueen(1, n);
        System.out.println("Solutions: " + resultCount);
    }
}
